.ci/run.sh

```diff
@@ -66,10 +66,6 @@ function run_whitespace() {
   fi
 }
 
-function run_style() {
```

Comment:
Is their any reason style doesn't do both things? We also end up spinning an extra VM instance on GitHub which is expensive and takes more time then just running all are checks in one thing. We can change the logic to not fail fast, so we get everything that's wrong over the fist set of errors from the first tool being executed.

Reply:
Unlike other linters, `isort` runs really quickly, yes. Style checking is not a bottle neck either way, so it doesn't really matter performance-wise. I find it more clean and organized to have one job per linter, especially if we want to add more linters in the future. That way you see instantly which linter failed and can instantly see the relevant log (instead of searching for the relevant bits).

Reply:
Its the VM spin-up time, nothing to do with whats executed. Also, we can only launch so many VMs at a time (offhand I think its 4). I'd rather dedicate parallel VM's to running actual function checks over additional "linter" checks. At some point in time, we will run multiple Python version checks before 1.0.

Right now we will have 3 nodes running just for style checks and if/when we add the linter, there's another instance.

What you can do is make each check it's own job, but run it in one node as multiple jobs. I think you can just mark the jobs as continue-on-error: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepscontinue-on-error

Reply:
Oh, that's a very good point! `continue-on-error` shows the stage as passed even though it failed, but I found another solution which continues on fail using `if: always()`.